Tests/test_SeqIO.py

```diff
@@ -4422,12 +4422,15 @@ def test_empty_file(self):
         """Check parsers can cope with an empty file."""
         for t_format in SeqIO._FormatToIterator:
             if t_format in SeqIO._BinaryFormats or \
-               t_format in ("uniprot-xml", "pdb-seqres", "pdb-atom", "cif-atom"):
+               t_format in ("uniprot-xml", "pdb-seqres", "pdb-atom", "cif-atom", "cif-seqres"):
                 # Not allowed empty SFF files.
-                continue
-            handle = StringIO()
-            records = list(SeqIO.parse(handle, t_format))
-            self.assertEqual(len(records), 0)
+                handle = StringIO()
```

Comment:
You shouldn't use ``StringIO`` on binary files, only plain text files. There's ``BytesIO`` for them. However, the test structure looks good.

i.e. If binary, use BytesIO empty test, elif non-empty file type, use StringIO for empty test, else parse file